tests/DCPS/Serializer/SerializerTest.cpp

```diff
@@ -695,6 +696,41 @@ bool runOverrunTest()
   return true;
 }
 
+bool runEncapsulationOptionsTest()
+{
+  std::cerr << "\nRunning encapsulation options tests...\n";
+  const CORBA::Octet arr[4] = {0};
+  const OpenDDS::DCPS::EncapsulationHeader encap;
+  OpenDDS::DCPS::unique_ptr<OpenDDS::DCPS::MessageBlockAllocator> mb_allocator_;
+
+  for (unsigned int i = 1; i <= OpenDDS::DCPS::EncapsulationHeader::padding_marker_alignment; i++) {
+    OpenDDS::DCPS::Message_Block_Ptr mb;
+    ACE_Message_Block* tmp_mb;
+
+    ACE_NEW_RETURN(tmp_mb,
+      ACE_Message_Block(4 + i, ACE_Message_Block::MB_DATA),
+      0);
+    mb.reset(tmp_mb);
+
+    OpenDDS::DCPS::Serializer serializer(mb.get(), encodings[5]);
+    serializer << encap;
+    serializer.write_octet_array(arr, i);
+
+    if (!OpenDDS::DCPS::EncapsulationHeader::set_encapsulation_options(mb)) {
+      std::cerr << "EncapsulationHeader::set_encapsulation_options failed" << std::endl << "size: " << mb->length();
```

Comment:
It would be helpful for the scoreboard if this was one line, also you're missing a new line at the end.